### ISSUE TITLE
vok-example-flow-sql2o - Build failed

### ISSUE THREAD

**Issue by jcgueriaud** [mvysny/vaadin-on-kotlin#7]
Hello,

It's not really a bug, I tried to build and run vok-example-flow-sql2o. It's not working.

I did this:

git clone https://github.com/mvysny/vaadin-on-kotlin
cd vaadin-on-kotlin
./gradlew vok-example-crud-sql2o:appRun

--> vok-example-crud-sql2o is working

then this:

./gradlew vok-example-flow-sql2o:appRun

And I got this:

18:24:13.895 ERROR [Bower] Install failed: ENOTEMPTY: Error: ENOTEMPTY:/home/jgueriaud/.cache/bower/packages/57f7f0864b724db288d6fd3200db7f1d/2.0.0
    at /home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1347
    at _fulfilled (/home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1346)
    at /home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1346
    at /home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1346
    at /home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1346
    at runSingle (/home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1346)
    at flush (/home/jgueriaud/dev_kotlin/vaadin-on-kotlin/vok-example-flow-sql2o/build/webResource/build.js:1346)
    at processImmediate (timers.js:345)

There is no kotlin problem but there is a problem with bower 

* Exception is:
org.gradle.api.tasks.TaskExecutionException: Execution failed for task ':vok-example-flow-sql2o:webResourceInstallBowerDependencies'.

Do you have the same error ? (and resolve it ? )

Thanks,

**Comment by mvysny** [comment]
Hi, thanks for letting me know! Yes, I tend to have the same bug from time to time. While I do not know what the exact cause is, there is a workaround: just try to build the whole VOK again, or just the `vok-example-flow-sql2o`, by invoking:

```bash
./gradlew vok-example-flow-sql2o:build
```

I currently believe that doing `./gradlew clean` increases the chance of this bug happening (since that would delete the `bower_components`), but this remains to be confirmed.

On my Travis this bug happens randomly.

So, in short, just try to run `./gradlew vok-example-flow-sql2o:build` until it succeeds, then you can safely run `./gradlew vok-example-flow-sql2o:build`

**Comment by mvysny** [comment]
Perhaps I should create a bug report, but I am totally at loss as of the cause, so I do not know whether the bug should be opened for Bower or the `com.github.ksoichiro.web.resource` Gradle plugin...

**Comment by jcgueriaud** [comment]
Thanks for your fast answer.

I tried multiple times but it's not working.

Finally I did (as a workaround):
bower install  (inside vok-example-flow-sql2o/build/webResource)

Then ./gradlew vok-example-flow-sql2o:build
And it's working.

I tried to clean the project then build it and I have the same error.

At least I can build the project and test it.

**Comment by mvysny** [comment]
Thanks for letting me know, and thanks for the workaorund! It seems to be a bug in the Gradle plugin then. I'll contact the author and discuss this with him further.

**Comment by jcgueriaud** [comment]
Ok.
Thanks a lot for your time.

**Comment by mvysny** [comment]
This should actually be fixed now that we no longer use bower: as per upstream Vaadin 10 moving towards webjars and using Maven BOM to sort out dependencies, VoK too moved to webjars.
